### PR TITLE
chore: adding CEL for psp-host-filesystem

### PR DESCRIPTION
**Pull request by JaydipGabani**
**What this PR does / why we need it**:

**Which issue(s) does this PR fix** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #541 

**Special notes for your reviewer**:

### CODE REVIEW

File: artifacthub/library/pod-security-policy/host-filesystem/1.1.0/samples/psp-host-filesystem/example_allowed.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-host-filesystem
+  labels:
+    app: nginx-host-filesystem-disallowed
```
Comment (maxsmythe) [inline artifacthub/library/pod-security-policy/host-filesystem/1.1.0/samples/psp-host-filesystem/example_allowed.yaml:6]: Do we need the app label? The allowed/disallowed is out of sync for a number of files.

File: library/pod-security-policy/host-filesystem/template.yaml [hunk 2]
```diff
@@ -39,110 +39,145 @@ spec:
...
+            expression: 'has(variables.anyObject.spec.containers) ? variables.anyObject.spec.containers : []'
+          - name: initContainers
+            expression: 'has(variables.anyObject.spec.initContainers) ? variables.anyObject.spec.initContainers : []'
+          - name: ephemeralContainers
+            expression: 'has(variables.anyObject.spec.ephemeralContainers) ? variables.anyObject.spec.ephemeralContainers : []'
+          - name: allContainers
+            expression: 'variables.containers + variables.initContainers + variables.ephemeralContainers'
+          - name: allowedHostPaths
+            expression: |
+              has(variables.params) ? (has(variables.params.allowedHostPaths) ? variables.params.allowedHostPaths : []) : []
```
Comment (maxsmythe) [inline library/pod-security-policy/host-filesystem/template.yaml:56]: nit: If you use !has() the chaining should look a bit cleaner and will end with the user-input variable value

File: library/pod-security-policy/host-filesystem/template.yaml [hunk 3]
```diff
@@ -39,110 +39,145 @@ spec:
...
+          - name: volumes
+            expression: |
+              variables.anyObject.spec.volumes.filter(volume, has(volume.hostPath))
+          - name: badHostPaths
+            expression: |
+              variables.volumes.filter(volume, 
+                (size(variables.allowedHostPaths) == 0) ||
+                !(variables.allowedHostPaths.exists(allowedHostPath, 
+                    volume.hostPath.path.startsWith(allowedHostPath.pathPrefix) && (
+                    !(allowedHostPath.readOnly == true) ||
```
Comment (maxsmythe) [inline library/pod-security-policy/host-filesystem/template.yaml:66]: nit: no need for `== true` in `allowedHostPath.readOnly == true` (already a bool), also we should allow for the possibility that the parameter will not be set (i.e. has() == false)

File: library/pod-security-policy/host-filesystem/template.yaml [hunk 4]
```diff
@@ -39,110 +39,145 @@ spec:
...
+                      (allowedHostPath.readOnly && !variables.allContainers.exists(c, 
+                      c.volumeMounts.exists(m, m.name == volume.name && !m.readOnly))))))
```
Comment (maxsmythe) [inline library/pod-security-policy/host-filesystem/template.yaml:68]: we should account for if readOnly is missing (has() == false) (both for the constraint parameter and for the VolumeMount object)

**Review by maxsmythe — CHANGES_REQUESTED** [review]

**Review by sozercan — APPROVED** [review]
LGTM

**Review by maxsmythe — APPROVED** [review]
LGTM

**Pull request merged**